python/MooseDocs/test/integrity/test_spec_files.py

```diff
@@ -37,9 +37,12 @@ def check(self, location):
                 messages.append("Missing a test spec file in '{}'".format(os.path.dirname(spec)))
         else:
             node = pyhit.load(os.path.join(location, 'tests'))
-            for block in moosetree.find(node, lambda n: n.name=='Tests'):
-                if block['type'] == 'PythonUnitTest':
-                    tested.add(block['input'])
+
+            # check for PythonUnitTest blocks in [Tests]
+            block = moosetree.find(node, lambda n: n.name=='Tests')
+            for subblock in moosetree.findall(block, lambda n: n):
+                if subblock['type'] == 'PythonUnitTest':
+                    tested.add(subblock['input'])
```

Comment:
At first I just did

```
for block in moosetree.find(block, lambda n: n.name='Tests'):
    if block['type'] == 'PythonUnitTest':
        tested.add(block['input'])
        continue

    for subblock in block:
        if subblock['type'] == 'PythonUnitTest':
            tested.add(subblock['input'])
```

Which was fine for the nested block structure we're looking for here. But then I figured we might as well assume that any subblock within [Tests] might be a python unit test, and to check an indefinite amount of nested levels. This way, as TestHarness and MOOSE HIT evolves in different ways, this should pretty much still work.